tests/test_cli.py

```diff
@@ -463,6 +463,25 @@ def test_whole_bibex(self):
                                                         full_cmd.split())
                         mocked_bib.assert_called()
 
+    def test_genetic_map_citations(self):
+        # Test that genetic map citations are converted.
+        species = stdpopsim.get_species("homsap")
+        genetic_map = species.get_genetic_map("HapmapII_GRCh37")
+        contig = species.get_contig("chr22", genetic_map=genetic_map.name)
+        model = stdpopsim.PiecewiseConstantSize(species.population_size)
+        engine = stdpopsim.get_default_engine()
+
+        with mock.patch("builtins.open", mock.mock_open()):
+            with open('tmp.bib', 'w') as bib:
+                with mock.patch.object(
```

Comment:
Why do we need two levels of patches here? Doesn't the first one catch them all?

Reply:
My thought was that I wanted to check to ensure that specifically the genetic map `fetch_bibtex` was being called, not just the general one (which we're already testing). But since I don't think its possible to check if a specific instantiation of the class, I separately patched out the genetic map `fetch_bibtex` and ensured that it gets called (while the previous ones still need to be patched to avoid actually downloading anything). We could alternatively just patch out everything and ensure that it gets called the correct number of times, but I thought this was more explicit. Up to you!

Reply:
Ah, I see. I think asserting it gets called the expected number of times would be good, or else add in a comment describing the strategy. These mocking tests can be  a bit opaque at times.